Src/library/net/Clib/network.c

```diff
@@ -824,81 +824,188 @@ void c_put_stream(EIF_INTEGER fd, EIF_POINTER stream_pointer, EIF_INTEGER length
 }
 
 
-EIF_REAL c_read_float (EIF_INTEGER fd, EIF_INTEGER *l_bytes)
-	/*x read a real from socket fd */
+/******************************************************/
+/* read operations                                    */
+/******************************************************/
+
+EIF_INTEGER c_recv_noexception(EIF_INTEGER fd, EIF_POINTER buf, EIF_INTEGER len, EIF_INTEGER flags)
```

Comment:
I would remove this routine, and keep the CREADX macro but simplify it to not perform the call to eif_net_check.